uber/config.py

```diff
@@ -1,124 +1,178 @@
 from uber.common import *
 
-# Any configurable property defined in our configuration is automatically converted into a global constant,
-# so outside of this file, we should never need to access this dictionary directly.  So we should prefer
-# using the DONATIONS_ENABLED global constant instead of saying _config['donations_enabled'], etc.  See
-# the comments in configspec.ini for explanations of the particilar options, which are documented there.
-# All global constants defined and exported here are also passed to our templates.
-_config = parse_config(__file__)
-
-class State:
-    """
-    We use global constants for all of our configurable global state.  This works really well for things
-    which do not change over time, such a the name of our event or what features are enabled.  However,
-    some things depend on the date/time (such as the badge price, which can change over time), or whether
-    we've hit our configured attendance cap (which changes based on the state of the database).  This
-    class offers properties and methods for checking the state of these things.  Much like our global
-    constants, the properties defined here are also passed directly to our templates.
-
-    There's a single global instance of this class called "state" which you should use, e.g. if you need
-    to check whether dealer registration is open in your code, you'd say state.DEALER_REG_OPEN (but in
-    a template you'd just say DEALER_REG_CLOSED).
 
```

Comment:
As you've seen, this class (and it's global ``c`` instance, which replaces ``state``) is now used everywhere instead of only for dynamic stuff.